// @mui-treasury/styles/esm/tabs/apple/appleTabs.styles.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { createStyles } from '@material-ui/core/styles';
export var tabsStyles = function tabsStyles(_ref) {
  var palette = _ref.palette,
      breakpoints = _ref.breakpoints;
  return createStyles({
    root: {
      backgroundColor: palette.type === 'light' ? '#eee' : palette.divider,
      borderRadius: 10,
      minHeight: 44
    },
    flexContainer: {
      display: 'inline-flex',
      position: 'relative',
      zIndex: 1
    },
    scroller: _defineProperty({}, breakpoints.up('md'), {
      padding: '0 8px'
    }),
    indicator: {
      top: 3,
      bottom: 3,
      right: 3,
      height: 'auto',
      background: 'none',
      '&:after': {
        content: '""',
        display: 'block',
        position: 'absolute',
        top: 0,
        left: 4,
        right: 4,
        bottom: 0,
        borderRadius: 8,
        backgroundColor: palette.type === 'light' ? '#fff' : palette.action.selected,
        boxShadow: '0 4px 12px 0 rgba(0,0,0,0.16)'
      }
    }
  });
};
export var tabItemStyles = function tabItemStyles(_ref2) {
  var palette = _ref2.palette,
      breakpoints = _ref2.breakpoints;
  return createStyles({
    root: _defineProperty({
      '&:hover': {
        opacity: 1
      },
      minHeight: 44,
      minWidth: 96
    }, breakpoints.up('md'), {
      minWidth: 120
    }),
    wrapper: {
      // zIndex: 2,
      // marginTop: spacing(0.5),
      color: palette.text.primary,
      textTransform: 'initial'
    }
  });
};
